vuu/src/main/scala/org/finos/vuu/core/filter/FilterTreeVisitor.scala

```diff
@@ -70,12 +70,26 @@ class FilterTreeVisitor extends FilterBaseVisitor[FilterClause] with StrictLoggi
 
   private def parseTypeAndValue(ctx: AtomContext): (Int, String) = {
     val atomType = ctx.getChild(0).asInstanceOf[TerminalNode].getSymbol.getType
-    val text = ctx.getText
-    (atomType, text)
+    val textValue = parseValue(atomType, ctx)
+
+    (atomType, textValue)
   }
 
   private def parseTypeAndValueList(ctx: List[AtomContext]): (Int, List[String]) = {
     val atomType = ctx.head.getChild(0).asInstanceOf[TerminalNode].getSymbol.getType
-    (atomType, ctx.map(ctx => ctx.getText))
+    val textValues = ctx.map(ctx => parseValue(atomType, ctx))
+
+    (atomType, textValues)
+  }
+
+  private def parseValue(atomType: Int, ctx: AtomContext) = {
+    atomType match {
+      case FilterLexer.STRING =>
```

Comment:
Would be nice if we could leave this to the parser - define a token for the string content (without the quotes) and define the grammar for a String as something like
```
'"' StringContent '"'
```
. Not sure if we can do it in such a way that it would not clash with the token for identifier, though.